core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -397,3 +397,31 @@ oppia.directive('testInteractionPanel', [function() {
     ]
   };
 }]);
+
+oppia.directive('heightWatcher', [function() {
+  return {
+    restrict: 'A',
+    link:function(scope, element, attrs) {
+    scope.$watch(
+      function() {
+        return element[0].offsetHeight;
+      },
+      function(newVal,oldVal) {
+        if (newVal!==oldVal) {
+          if (newVal>630 - 40) {
+            if(angular.element(element[0]).find("#heightchecker").length == 0) {
+              angular.element(element[0]).append(
```

Comment:
You'll need to find another way to do this. We don't construct code by concatenating strings in JS; it needs to be done in the templates.